### ISSUE TITLE
Mother 3 - Constant audio clicking

### ISSUE THREAD

**Issue by ghost** [mgba-emu/mgba#461]
I dunno if it's me, but I tend to get audio clicking when playing this game using the default sample rate settings.

Lowering it to it's lowest fixes it a bit, but results in more compressed audio. Here's a MP3 sample of the title screen music, which has the most major clicking from what I hear.

[Mother 3 MGBA clicking examples.zip](https://github.com/mgba-emu/mgba/files/578086/Mother.3.MGBA.clicking.examples.zip)

**Comment by endrift** [comment]
Have you tried increasing the buffer size?

**Comment by TheGoatManfan** [comment]
I did, it was set to max, the clicks still happen during the title screen. And some other clicks in other game music. It's not all the time, but it's noticeable.

**Comment by endrift** [comment]
I can't reproduce this on my end. Which version are you using?

**Comment by Alazu** [comment]
I also get this audio clicking bug... in the Mother 3 Fan translation.... But not as Harshly, also a Lower bitrate of Audio.... Seems Much Better in VBA and M-VBA Sucks with Audio emulation.... Oh well... this Seems like a great project though! I don't know if I tried in Any other emulator  i.e RetroArch's Emulation... Oh wait! MGBA is Core process in RetroArch! Duh....

**Comment by CharlesCorrin** [comment]
I also hear some audio clicking on the title screen, around every couple of seconds. I hear it in every build I try (I like to keep the emulator updated and Mother 3 is my test game to be sure it works).

But get this: I also hear the clicks more if I set the framerate to native than if I set it to 60. On 60, the clicks are much less often and not as loud. What's your framerate set to, TheGoatManfan?

**Comment by Yuki-nyan** [comment]
This is much more noticeable on the Vita port. I've taken two recordings of track 58 from the sound test on the Windows and Vita ports to show the difference: http://de-ge.so/vita/misc/mother3.zip

This track seems to play fine on Windows with the default buffer size although I do still experience the clicking on the title screen.

**Comment by CharlesCorrin** [comment]
Take note please: At some point my PC stopped behaving differently on different FPS settings. For whatever reason (hardware change? Sound options change? Don't know...). Now both settings pop the same as the 60 FPS setting did before (and it was the option with less popping). Which is not too bad, but still noticeable. I still use the Mother 3 title screen for testing.

**Comment by endrift** [comment]
I've never been able to reproduce this on macOS. I'll try on Windows 10.

**Comment by endrift** [comment]
Seems to be an audio issue that only affects Windows. Possibly related to SDL.

**Comment by GorgWorgington** [comment]
I'm experiencing this bug on Manjaro Linux. The easiest way to replicate is to talk to a frog to save your game. As soon as the save menu comes up, a noticeable crackle will play at the very beginning of the music track.

**Comment by NarryG** [comment]
Experiencing the save game variation as well, but I'm not really getting any popping anywhere else. Might be a separate bug?

**Comment by CharlesCorrin** [comment]
I have an update for this. I did some real testing this time.

If I set the audio buffer to 1536 (the default), the Mother 3 title screen pops like crazy. The same happens when I set it to 512, 768, or 3072.

If I set the audio buffer to 1024, 2048, or 4096, the popping is gone. I don't know why this is, but it is.

I recommend everyone play around with the audio buffer and see what works best for you. I'm embarrassed I didn't try this sooner.
